api/grpc/server/server.go

```diff
@@ -56,6 +57,9 @@ func (s *apiServer) CreateContainer(ctx context.Context, c *types.CreateContaine
 	e.Runtime = c.Runtime
 	e.RuntimeArgs = c.RuntimeArgs
 	e.StartResponse = make(chan supervisor.StartResponse, 1)
+	if runt.GOOS == "solaris" && len(c.Id) >= 12 {
```

Comment:
We tried putting it in the runtime but that didn't work out. I'll definitely factor it out into a function.